src/api/v1/users/register/volunteers.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Registration endpoints for volunteers
+ */
+import * as Hapi from 'hapi';
+import * as Boom from 'boom';
+import * as Joi from 'joi';
+import {
+  id,
+  userName,
+  birthYear,
+  email,
+  phoneNumber,
+  postCode,
+  password,
+  isEmailConsentGranted,
+  isSMSConsentGranted,
+  gender,
+  response
+} from '../schema';
+import {
+  Volunteers,
+  Users,
+  CommunityBusinesses
+} from '../../../../models';
+import { RoleEnum } from '../../../../auth/types';
+import { VolunteerRegisterRequest } from '../../types';
+
+export default [
+  {
+    method: 'POST',
+    path: '/users/register/volunteers',
+    options: {
+      description: 'Retreive list of all users',
+      auth: false,
+      validate: {
+        payload: {
+          organisationId: id.required(),
+          role: Joi.alternatives(RoleEnum.VOLUNTEER, RoleEnum.VOLUNTEER_ADMIN),
+          adminCode: Joi.string().regex(/^[0-9]{5}$/),
```

Comment:
This doesn't support the existing admin codes. Also lets leave ourselves a little leeway to support more complex codes.

Reply:
Can you clarify?

Reply:
Existing admin codes are alphanumeric and longer than 5 characters (I think 8). The bit about more complex codes was about allowing symbols and such, but actually that's probably annoying for users who have to enter these codes, so ignore that part.